services/finalizer/standard/handler.go

```diff
@@ -484,9 +484,6 @@ func (s *Service) fetchBlock(ctx context.Context, root phase0.Root) (*chaindb.Bl
 			return nil, nil
 		}
 
-		// We need to ensure the finalizer is not running ahead of the blocks service.  To do so, we compare the slot of the block
```

Comment:
Not sure why this comment has been removed.  It appears to accurately describe the inter-module situation and the purpose of the code following.  Please could you explain your rationale here?

Reply:
The finalizer was ~2000 slots ahead of the block service (`finalizer.standard` vs `blocks.standard`), which I didn't expect. This has happened a couple of times. 
The code will return `nil` or an error if a) there is a fetching error in `LatestBlocks` b) there are no blocks c) the block version is earlier than the phase it is supposed to be
In all other cases it is execute `OnBlock`. Based on the comment, I was expecting to compare `signedBlock` with `latestBlocks[0]` and return.
I may be missing something?

Reply:
That sounds like a separate issue.  If the finalizer runs ahead of the blocks module it is possible for the blocks module to reset the finality information, which wouldn't be great.  We should keep the comment and address the issue that caused the finalizer to get ahead.